driver/src/orderbook/streamed/block_timestamp.rs

```diff
@@ -21,3 +21,36 @@ impl BlockTimestamp for Web3 {
         .boxed()
     }
 }
+
+/// A cache for the block timestamp which avoids having to query the node in the case where we
+/// receive multiple events from the same block in a row.
+#[derive(Debug)]
+pub struct BlockTimestampCache<T> {
+    block_timestamp: T,
```

Comment:
To me a `block_timestamp` is an already materialized concrete timestamp for a block not a component that allows querying the block_timestamp (given that T can be anything here it's not clear from the type either).
Can we maybe call this `block_timestamp_reader` or even more abstract `cache_miss_reader` or the like?

Reply:
or even just calling it `inner` would be helpful IMO